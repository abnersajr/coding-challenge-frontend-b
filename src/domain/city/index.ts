export * as CityDomain from './City'
