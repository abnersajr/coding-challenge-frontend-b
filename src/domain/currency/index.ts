export * as CurrencyDomain from './Currency'
