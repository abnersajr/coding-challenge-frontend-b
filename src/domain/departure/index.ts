export * as DepartureDomain from './Departure'
