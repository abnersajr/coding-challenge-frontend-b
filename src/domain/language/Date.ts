import { format, formatISO, startOfToday } from 'date-fns'
import { LanguageDomain } from '.'

export const localeDateMask = (locale: LanguageDomain.Language) => {
  switch (locale) {
    case LanguageDomain.PT:
      return 'dd/MM/yyyy'
      break

    default:
      return 'MM/dd/yyyy'
      break
  }
}
export const dateFormatted = (date: Date, mask: string = 'yyyy-MM-dd') => {
  return format(date, mask)
}

export const dateISO = (date: Date) => formatISO(date)

export const todayString = () => dateISO(new Date())
