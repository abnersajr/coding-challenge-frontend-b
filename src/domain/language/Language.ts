import { Language } from 'grommet-icons'

export const EN = 'en'
export const FR = 'fr'
export const ES = 'es'
export const PT = 'pt'

type LanguageCode = typeof EN | typeof FR | typeof ES | typeof PT
export interface Language {
  readonly code: LanguageCode
}
