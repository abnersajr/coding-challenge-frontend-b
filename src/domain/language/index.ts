export * as LanguageDomain from './Language'
export * as DateDomain from './Date'
