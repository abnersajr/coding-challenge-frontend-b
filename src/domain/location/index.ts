export * as LocationDomain from './Location'
