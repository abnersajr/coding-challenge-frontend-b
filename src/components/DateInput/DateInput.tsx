import React from 'react'
import { useRouter } from 'next/router'
import { IntlText } from '../Intl/IntlText'
import { FormField, DateInput as Input, DateInputProps } from 'grommet'
import { DateDomain, LanguageDomain } from '../../domain/language'

type Props = {
  value?: string
  onChange: DateInputProps['onChange']
}
const DateInput = (props: Props) => {
  const { locale } = useRouter()
  const format = DateDomain.localeDateMask(locale as LanguageDomain.Language)

  const dateField = {
    readOnly: true,
    calendarProps: {
      'data-testid': 'CALENDAR',
      locale: locale,
      size: 'small',
      bounds: [DateDomain.todayString(), '2025-01-01'],
    },
    name: 'date',
    defaultValue: props.value || DateDomain.todayString(),
  }

  return (
    <FormField width="150px" label={<IntlText id="date" />}>
      <Input
        data-testid="INPUT.DATE"
        format={format}
        {...dateField}
        onChange={props.onChange}
      />
    </FormField>
  )
}

export default DateInput
