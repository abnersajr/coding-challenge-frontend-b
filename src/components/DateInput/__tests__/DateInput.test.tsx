import React from 'react'
import DateInput from '../DateInput'
import { DateDomain } from '../../../domain/language'
import { render, screen, fireEvent } from 'test/components'

describe('DateInput', () => {
  const todaySplit = new Date('10/22/2022')
    .toISOString()
    .split('T')[0]
    .split('-')
  const today = `${parseInt(todaySplit[1])}/${todaySplit[2]}/${todaySplit[0]}`

  it('renders the input with actual date', () => {
    const wrapper = render(
      <DateInput value={today} onChange={(event) => jest.fn()} />
    )
    expect(wrapper.getByTestId('INPUT.DATE')).toHaveValue(today)
  })

  it('matchs the snapshot', () => {
    const wrapper = render(
      <DateInput value={today} onChange={(event) => jest.fn()} />
    )
    expect(wrapper.getByTestId('INPUT.DATE')).toMatchSnapshot()
  })
})
