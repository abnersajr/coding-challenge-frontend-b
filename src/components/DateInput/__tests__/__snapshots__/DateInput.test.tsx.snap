// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateInput matchs the snapshot 1`] = `
<input
  autocomplete="off"
  class="StyledMaskedInput-sc-99vkfa-0 jLezbT"
  data-testid="INPUT.DATE"
  name="date"
  placeholder="MM/dd/yyyy"
  readonly=""
  value="10/22/2022"
/>
`;
